framework/src/kernels/ADKernelScalarBase.C

```diff
@@ -77,6 +77,9 @@ ADKernelScalarBase::computeJacobian()
 {
   ADKernel::computeJacobian();
 
+#ifndef MOOSE_SPARSE_AD
+  mooseError("ADKernelScalarBase assembly only supported for non-sparse AD");
+#else
```

Comment:
If you have already established the restriction that these objects can only run with global AD indexing, then it's not necessary to guard against nonsparse AD. Global AD indexing only works with an underlying sparse container

Reply:
But I understand why you're doing this in order to make the nonsparse recipe compile. But I'm going to modify `Assembly` such that you don't have to do these guards in residual object code